test/cpp/api/transformer.cpp

```diff
@@ -0,0 +1,334 @@
+                                 tensor_options);
+  memory_input = torch::tensor({{{9, 10, 11, 12}},
+                                {{11, 12, 13, 14}}},
+                                tensor_options);
+  result = model(decoder_input, memory_input).detach();
+  ref_output = torch::tensor({{{2.338531, 0.087709, -0.65776, 0.080646}},
+                              {{2.338531, 0.087709, -0.65776, 0.080646}}},
+                              tensor_options);
+  ASSERT_EQ(result.sizes().size(),ref_output.sizes().size());
+  ASSERT_TRUE(torch::allclose(result, ref_output, 1e-7, 1e-5,
+                              /*equal_nan=*/true));
+
+
+  // deterministic input
+  decoder_input = torch::tensor({{{0.4517, 0.6793, 0.5313, 0.0034},
+                                  {0.2678, 0.3677, 0.4459, 0.7166}},
+                                 {{0.8100, 0.3716, 0.4096, 0.1976},
+                                  {0.6958, 0.8844, 0.6081, 0.8315}},
+                                 {{0.0494, 0.9343, 0.5955, 0.3830},
+                                  {0.5404, 0.3464, 0.9378, 0.6200}}},
+                                 tensor_options);
+  memory_input = torch::tensor({{{0.7462, 0.6653, 0.5679, 0.4891},
+                                 {0.5387, 0.1655, 0.3565, 0.0471}},
+                                {{0.8335, 0.2799, 0.5031, 0.2947},
+                                 {0.1402, 0.0318, 0.7636, 0.1346}},
+                                {{0.6333, 0.9344, 0.1376, 0.9938},
+                                 {0.8924, 0.2872, 0.6692, 0.2944}},
+                                {{0.9897, 0.6915, 0.3154, 0.1733},
+                                 {0.8645, 0.3513, 0.3064, 0.0767}},
+                                {{0.8117, 0.2366, 0.4838, 0.7881},
+                                 {0.3718, 0.4945, 0.9511, 0.0864}}},
+                                tensor_options);
+  result = model(decoder_input, memory_input).detach();
+  ref_output = torch::tensor(
+    {{{2.42049104, 0.03443088, -0.60793706, -0.05436271},
+     {2.42210631, 0.03546578, -0.60679895, -0.05357488}},
+    {{2.41907674, 0.0336104, -0.60892977, -0.05490462},
+     {2.42216881, 0.03586554, -0.6067524, -0.05289126}},
+    {{2.42205716, 0.03488046, -0.60683681, -0.05460596},
+     {2.42240309, 0.0354595, -0.60659063, -0.05378816}}},
+    tensor_options);
+  ASSERT_EQ(result.sizes().size(),ref_output.sizes().size());
+  ASSERT_TRUE(torch::allclose(result, ref_output, 1e-7, 1e-5,
+                              /*equal_nan=*/true));
+
+}
+
+TEST_F(TransformerTest, TransformerDecoderLayer_gelu) {
+  transformer_decoder_layer_test_helper_gelu(false);
+}
+
+TEST_F(TransformerTest, TransformerDecoderLayer_geluCUDA) {
+  // Check to prevent execution on non CUDA devices
+  if (torch::cuda::is_available()){
+    transformer_decoder_layer_test_helper_gelu(true);
+  }
+
+}
+
+TEST_F(TransformerTest, PrettyPrintTransformerDecoderLayer) {
+  ASSERT_EQ(
+      c10::str(TransformerDecoderLayer(4, 2)),
```

Comment:
By default contained sub-modules are printed. The implementation is commented since it was printing duplicate values
UT is successful as below
[ RUN      ] TransformerTest.PrettyPrintTransformerDecoderLayer
[       OK ] TransformerTest.PrettyPrintTransformerDecoderLayer (1 ms)

Reply:
Default output with sub modules is being verified here. When pretty print is implemented in python, this can be updated to be the same.
now the UT is successful
[ RUN      ] TransformerTest.PrettyPrintTransformerDecoderLayer
[       OK ] TransformerTest.PrettyPrintTransformerDecoderLayer (1 ms)

Reply:
@VinodSKumar 
Got u, u are right, I made a mistake. Thanks